source/s3_meta_request.c

```diff
@@ -610,6 +610,20 @@ static void s_s3_meta_request_sign_request(
     meta_request->vtable->sign_request(meta_request, request, on_signing_complete, user_data);
 }
 
+static bool s_is_put_request(const struct aws_s3_meta_request *meta_request) {
+    if (meta_request->type == AWS_S3_META_REQUEST_TYPE_PUT_OBJECT) {
+        return true;
+    }
+    if (meta_request->type == AWS_S3_META_REQUEST_TYPE_DEFAULT) {
+        struct aws_byte_cursor method;
+        int err = aws_http_message_get_request_method(meta_request->initial_request_message, &method);
```

Comment:
Feels like you could simplify this to something like:

```
if (!aws_http_message_get_request_method(...)) {
   return aws_byte_cursor_eq(...);
}
```
and let the fall through the hit the return false;